### PR TITLE
Run CI pipeline on pull requests too

### PR DESCRIPTION
**Pull request by academo**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -1,13 +1,18 @@
 name: Node CI
```
Comment (academo) [inline .github/workflows/test.yml:1]: I removed all the mentions about tokens here.

**Review by andresmgot — APPROVED** [review]
Changes makes sense from the GH action perspective :+1: Maybe wait for @jackw to review the release content.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
+name: Node CI
+on:
+  push:
+    branches:
+      - main
+  pull_request_target:
```
Comment (jackw) [inline .github/workflows/release.yml:6]: Are we sure we want to allow npm releases from forks? Doesn't `pull_request_target` allow an external contributor to release a canary package with any code they like in it. 🤔

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        with:
+          node-version: '20'
+          registry-url: 'https://registry.npmjs.org'
+          cache: 'npm'
+
+      - name: Install dependencies
+        run: npm ci --no-audit
+
+      - name: Build
+        run: npm run build
```
Comment (jackw) [inline .github/workflows/release.yml:46]: If I fork this repo can I change this npm script in a way that will allow me to steal `process.env.NPM_TOKEN`?

File: .github/workflows/test.yml (hunk 1 above)
Comment (jackw) [reply .github/workflows/test.yml:1]: `secrets.GRAFANA_ACCESS_POLICY_TOKEN`, and `secrets.GRAFANA_API_KEY` appear to still exist in this file. Is that an issue?

File: .github/workflows/release.yml (hunk 3 above)
Comment (academo) [reply .github/workflows/release.yml:46]: Good point. I think it is just simpler to leave pull_request here

File: .github/workflows/test.yml (hunk 1 above)
Comment (academo) [reply .github/workflows/test.yml:1]: I removed all the code that used those tokens. Can you point me where they are mentioned?

**Comment by academo** [comment]
GRAFANA_API_KEY

**Pull request closed without merging**